common/djangoapps/student/models.py

```diff
@@ -537,6 +537,7 @@ def activate(self):
         self.user.is_active = True
         self._track_activation()
         self.user.save()
+        self.referral_uid = uuid.uuid4().hex
```

Comment:
@RuslanSkiraRaccoongang please use `get_user_id()` ([commit](https://github.com/raccoongang/edx-platform/commit/558071c0c5c46e465da5b3e88eeba29a08add473#diff-f2bb10e5e8eb41e991aff45d799e4de5R61)) for id to be consistent

Reply:
@OPersian  check again please.